rosbag2_tests/test/temporary_directory_fixture.hpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TEMPORARY_DIRECTORY_FIXTURE_HPP_
+#define TEMPORARY_DIRECTORY_FIXTURE_HPP_
+
+#include <gmock/gmock.h>
+
+#include <string>
+
+#ifdef _WIN32
+# include <direct.h>
+# include <Windows.h>
+#else
+# include <unistd.h>
+# include <sys/types.h>
+# include <dirent.h>
+#endif
+
+using namespace ::testing;  // NOLINT
+
+class TemporaryDirectoryFixture : public Test
```

Comment:
maybe it's just the naming, but why is this temporary?

Reply:
It gets deleted at destruction. This fixes the issue you have seen where folders get created and not cleaned up correctly. That's why I thought the "temporary" is appropriate. Of course, I'm happy to change the name if a better one is suggested :-)